presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestJoinQueries.java

```diff
@@ -912,6 +912,19 @@ public void testJoinWithInSubqueryToBeExecutedAsPostJoinFilter()
         }
     }
 
+    @Test
+    public void testOuterJoinWithExpression()
+    {
+        assertQuery("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey = o.orderkey");
```

Comment:
How are these testing dynamic filters?

Reply:
AbstractTestJoinQueries testcases will be triggered in 
TestHiveDistributedJoinQueriesWithDynamicFiltering and 
TestHiveDistributedJoinQueriesWithDynamicFilteringAndFilterPushdown
and many other tests, to guarantee the correctness